### PR TITLE
Exertion effects now consider tool quality part 2: shovels

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Balance "Tool quality and other time-saving factors now impact fatigue and other needs penalties from digging"

#### Purpose of change

A belated followup to https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2755 and https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2617 that I'd been meaning to do, this sets it so that the fake-exertion effects from digging pits is naturally calculated from time taken to dig, thus making it so that better tools, more strength, better helpers etc all modify how much fatigue, thirst, etc digging inflicts on you.

~~WIP because I need to also apply this to the other digging-related functions in the relevant file, re-test, then check for astyle, but wanted to make sure the basic method worked first then test something else right quick.~~ And took care of all that.

#### Describe the solution

In activity_actor.cpp, set it to define the now-familiar act_exertion int using the total_moves of the assigned digging activity, and have it calculate kcal reduction, thirst increment, and fatigue increment using it. This works in the exact same way as the calculations used in woodcutting since both activities treat 60 minutes as the baseline time. Applied this for finishing both digging pits and for the related channeling function.

Main difference is I figured out I can use the total moves of the activity instead of having to recreate chop_moves like I did in the previous PR. This not only saves me some redundant definitions in the finisher function, but also means not having to assign the tool used for the activy to look up later, thus avoiding problems like https://github.com/cataclysmbnteam/Cataclysm-BN/issues/3015 wherein I fucked up and forgot that the NPC version of the activity needs to remember the item used too.

Next, in activity_handlers.cpp, applied another change to the exertion functions of. This one had to be done a bit different too, since total_moves isn't conveniently already defined for me here. On top of that, time taken to fill in pits and tamp ground varies based off terrain filled in instead of tool used. I could tweak this to make tools affect it, but the time valued used are much smaller (peak of 15 minutes) compared to digging pits, so I figured it's reasonable to leave it for now.

#### Describe alternatives you've considered

1. Backporting this use of `moves_total` to the woodcutting function for yet another PR messing with it.
2. Porting DDA's exertion mechanics but streamlining them to be less awful to work with so I can skip all this nonsense entirely. Even then I'd likely need to hack in some kinda extra bit to make tool qualities matter as I highly doubt DDA makes anything less tiring to do if you have better tools.
3. Reworking filling in pits to consider tool quality used, likely in exchange for bumping up the time costs a bit more, in particular making the worst-case scenario of filling in a deep pit with a digging stick (funny enough the game lets you do this even just fine) closer to 30 minutes.
4. Making it so that leveling off mounds of dirt grants no fatigue, on the basis that making said mounds doesn't inflict fatigue.

#### Testing

1. Compiled and load-tested.
2. Set debug needs mutation on, zeroed out needs, set strength to 10, and spawned in a digging stick
5. Dug a shallow pit, confirmed it took an hour, confirmed it inflicted 10 fatigue afterward as the expected baseline, thirst gain was 5, etc.
6. Repeated test with a stone shovel, fatigue gain was 5 as expected, other needs affected accordingly.
7. Repeated test with a full-on shovel, fatigue gain was 3, other needs affected accordingly.
8. Tested filling in a deep pit, gained 10 fatigue.
9. Tested filling in a shallow pit, gained 6 fatigue.
10. Tested filling in a dirt mound, gained 3 fatigue.
11. Checked affected files for astyle.

### CODE REVIEW

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -455,10 +455,11 @@ void dig_activity_actor::finish( player_activity &act, Character &who )
                                   calendar::turn ) );
     }
 
-    const int helpersize = character_funcs::get_crafting_helpers( who, 3 ).size();
-    who.mod_stored_nutr( 5 - helpersize );
-    who.mod_thirst( 5 - helpersize );
-    who.mod_fatigue( 10 - ( helpersize * 2 ) );
+    int act_exertion = act.moves_total;
```
Comment (scarf005) [inline src/activity_actor.cpp:458]:
```suggestion
    const int act_exertion = act.moves_total;
```
To make the code easier to understand, it'd be great if all variable are declared immutable (const) unless there's a compelling reason to mutate it.

**Review by scarf005 — CHANGES_REQUESTED** [review]

**Review by Firestorm01X2 — APPROVED** [review]
Ready

**Comment by chaosvolt** [comment]
Pffft, requested change didn't update even after committing the code suggestion in it.

File: src/activity_actor.cpp [hunk 2]
```diff
@@ -455,10 +455,11 @@ void dig_activity_actor::finish( player_activity &act, Character &who )
...
 
-    const int helpersize = character_funcs::get_crafting_helpers( who, 3 ).size();
-    who.mod_stored_nutr( 5 - helpersize );
-    who.mod_thirst( 5 - helpersize );
-    who.mod_fatigue( 10 - ( helpersize * 2 ) );
+    const int act_exertion = act.moves_total;
+
+    who.mod_stored_kcal( std::min( -1, -act_exertion / to_moves<int>( 80_seconds ) ) );
+    who.mod_thirst( std::max( 1, act_exertion / to_moves<int>( 12_minutes ) ) );
+    who.mod_fatigue( std::max( 1, act_exertion / to_moves<int>( 6_minutes ) ) );
```
Comment (olanti-p) [inline src/activity_actor.cpp:462]: NPC assistants are no longer considered. Intentional?

Comment (chaosvolt) [reply src/activity_actor.cpp:462]: Exertion effects scale with the time taken to complete the activity, and helpers already reduce that in: https://github.com/cataclysmbnteam/Cataclysm-BN/blob/upload/src/iuse.cpp#L2725

**Review by scarf005 — APPROVED** [review]

**Pull request merged**